File: lib/layer2_core/light/e_on_background_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class EOnBackgroundLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl2;
      case 'standard': return cstl2;
      case 'blue': return cbl2;
      case 'green': return cgl2;
      case 'red': return crl2;
      case 'nothing': return cnol2;
      default: return cstl2;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/EOnBackgroundLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnBackgroundLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl2
            AppTheme.STANDARD -> R.color.cstl2
            AppTheme.BLUE -> R.color.cbl2
            AppTheme.GREEN -> R.color.cgl2
            AppTheme.RED -> R.color.crl2
            AppTheme.NOTHING -> R.color.cnol2
        }
    }
}
